pkg/pipelines/argocd/argocd.go

```diff
@@ -38,46 +38,49 @@ func Build(argoNS, repoURL string, m *config.Manifest) (res.Resources, error) {
 	if repoURL == "" {
 		return res.Resources{}, nil
 	}
-	argoEnv, err := m.GetArgoCDEnvironment()
-	// If there's no ArgoCD environment, then we don't need to do anything.
-	if err != nil {
+	argoEnv := m.GetArgoCDConfig()
+	if argoEnv == nil {
 		return res.Resources{}, nil
 	}
 
 	files := make(res.Resources)
 	eb := &argocdBuilder{repoURL: repoURL, files: files, argoEnv: argoEnv, argoNS: argoNS}
-	err = m.Walk(eb)
+	err := m.Walk(eb)
 	return eb.files, err
 }
 
 type argocdBuilder struct {
 	repoURL string
-	argoEnv *config.Environment
+	argoEnv *config.Argo
 	files   res.Resources
 	argoNS  string
 }
 
 func (b *argocdBuilder) Application(env *config.Environment, app *config.Application) error {
-	basePath := filepath.Join(config.PathForEnvironment(b.argoEnv), "config")
+	basePath := filepath.Join(config.PathForArgoEnvironment(), "config")
 	argoFiles := res.Resources{}
 	filename := filepath.Join(basePath, env.Name+"-"+app.Name+"-app.yaml")
 	argoFiles[filename] = makeApplication(env.Name+"-"+app.Name, b.argoNS, defaultProject, env.Name, defaultServer, makeSource(env, app, b.repoURL))
 	b.files = res.Merge(argoFiles, b.files)
+	err := argoEnvironmentResources(b.argoEnv, b.files)
```

Comment:
It's argocd.  Or you mean https://en.wikipedia.org/wiki/Argo_(2012_film)

Reply:
Now changed to ArgoCDConfig and PipelineConfig